test/scales/linearScaleTests.ts

```diff
@@ -1,134 +1,289 @@
+        assert.strictEqual(scale.invert(500), 10, "last value in range maps to last value in flipped domain");
+        assert.strictEqual(scale.invert(450), 15, "middle value in range maps to middle value in flipped domain");
+      });
+
+      it("filters out invalid numbers when using extentOfValues()", () => {
+        let arrayWithBadValues: any[] = [null, NaN, undefined, Infinity, -Infinity, "a string", 1, 1.2];
+        let expectedExtent = [1, 1.2];
+        let extent = scale.extentOfValues(arrayWithBadValues);
+        assert.deepEqual(extent, expectedExtent, "invalid values were filtered out");
+      });
+
+      it("does not accept NaN or Infinity as domain", () => {
+        let warningCalled = false;
+        let oldWarn = Plottable.Utils.Window.warn;
+        Plottable.Utils.Window.warn = (msg: string) => {
+          if (msg.indexOf("NaN or Infinity") > -1) {
+            warningCalled = true;
+          }
+        };
+
+        scale.domain([1, 2]);
+        assert.deepEqual(scale.domain(), [1, 2], "initial domain is set");
+
+        warningCalled = false;
+        scale.domain([5, Infinity]);
+        assert.deepEqual(scale.domain(), [1, 2], "Infinity containing domain was ignored");
+        assert.isTrue(warningCalled, "a warning was thrown for setting domain to Infinity");
+
+        warningCalled = false;
+        scale.domain([-Infinity, 5]);
+        assert.deepEqual(scale.domain(), [1, 2], "-Infinity containing domain was ignored");
+        assert.isTrue(warningCalled, "a warning was thrown for setting domain to -Infinity");
+
+        warningCalled = false;
+        scale.domain([NaN, 7]);
+        assert.deepEqual(scale.domain(), [1, 2], "NaN containing domain was ignored");
+        assert.isTrue(warningCalled, "a warning was thrown for setting domain to NaN");
+
+        warningCalled = false;
+        scale.domain([-1, 5]);
+        assert.deepEqual(scale.domain(), [-1, 5], "Regular domains still accepted");
+        assert.isFalse(warningCalled, "a warning is not thrown when correct domains are given");
+
+        Plottable.Utils.Window.warn = oldWarn;
+      });
+
+      it("accepts NaN or Infinity as range", () => {
+        scale.range([5, Infinity]);
+        assert.deepEqual(scale.range(), [5, Infinity], "Infinity accepted as part of range");
+        scale.range([-Infinity, 5]);
+        assert.deepEqual(scale.range(), [-Infinity, 5], "-Infinity accepted as part of range");
+        scale.range([NaN, 7]);
+        assert.deepEqual(scale.range(), [NaN, 7], "NaN accepted as part of range");
+      });
+
+      it("can receive custom tick generator", () => {
+        scale.domain([0, 10]);
+        let defaultTicks = scale.ticks();
+        assert.strictEqual(defaultTicks.length, 11, "ticks were generated correctly with default generator");
+        assert.deepEqual(defaultTicks, Array.apply(null, Array(11)).map((d: any, i: number) => i),
```

Comment:
I think it would be clearer if the array from 0 to 10 was written out explicitly:
```Javascript
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
```